src/file/file.service.ts

```diff
@@ -265,17 +267,16 @@ export class FileService {
    * @param query - the conditions.
    * @param currArray - the array sent in the recursion.
    */
-  private static async getPopulatedChildren_aux(folderID: string, ownerID: string, query: Partial<IFile>) : Promise<ResFile[]> {
+  private static async getPopulatedChildren_recursive(folderID: string, ownerID: string, query: Partial<IFile>) : Promise<ResFile[]> {
     const folderFiles: IFile[] = await this.getFilesByFolder(folderID, ownerID, query);
     const childrenArray : ResFile[] = [];
 
-    // use the recursion to get the file's populated children
-    for (let i = 0 ; i < folderFiles.length ; i++) {
-      const currChild = new ResFile(folderFiles[i]);
-      const grandChildren = await this.getPopulatedChildren_aux(folderFiles[i].id, ownerID, query);
+    await Promise.all(folderFiles.map(async (child) => {
```

Comment:
I didn't notice it is an async function, a for loop seems to be a more readable option.
Resolved face to face.